ngraph/core/reference/include/ngraph/runtime/reference/function.hpp

```diff
@@ -0,0 +1,34 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+#include <vector>
+#include "ngraph/function.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            HostTensorVector function(const std::shared_ptr<Function>& function,
```

Comment:
I guess we will use it in a `evaluate `method of ngraph::function class, so reference has exactly the same name.
@ilyachur @GlebKazantaev what do you think?

Reply:
It is not a public API. And I think we can use `function` as a name because namespace `reference` should provide more information about purpose of this method.